test/integration/infra.go

```diff
@@ -78,27 +81,44 @@ func (infra *infra) setup() error {
 		}
 	}
 
-	deploy := func(name string) error {
+	if infra.IstioNamespace == "" {
+		var err error
+		if infra.IstioNamespace, err = util.CreateNamespace(client); err != nil {
+			return err
+		}
+		infra.istioNamespaceCreated = true
+	} else {
+		if _, err := client.Core().Namespaces().Get(infra.IstioNamespace, meta_v1.GetOptions{}); err != nil {
+			return err
+		}
+	}
+
+	deploy := func(name, namespace string) error {
 		if yaml, err := fill(name, infra); err != nil {
 			return err
-		} else if err = infra.kubeApply(yaml); err != nil {
+		} else if err = infra.kubeApply(yaml, namespace); err != nil {
 			return err
 		}
 		return nil
 	}
-	if err := deploy("rbac-beta.yaml.tmpl"); err != nil {
+	if err := deploy("rbac-beta.yaml.tmpl", infra.IstioNamespace); err != nil {
 		return err
 	}
-	if err := deploy("config.yaml.tmpl"); err != nil {
+	if err := deploy("config.yaml.tmpl", infra.Namespace); err != nil {
```

Comment:
shouldn't this be IstioNamespace as well?

Reply:
The config map must be deployed in the app namespaces, as well as in the istio namespace, istio services read it as well, or we may have apps deployed in the istio namespace.

Reply:
Config map is picked up at startup by proxy agents. Can kubernetes mount a configmap from another namespace?